contracts/stake/src/state.rs

```diff
@@ -282,101 +297,106 @@ impl StakeState {
         STAKE_CONTRACT_VERSION
     }
 
-    /// Slash the given `to_slash` amount from a `stake_pk` reward
+    /// Slash the given `to_slash` amount from a `account` reward
```

Comment:
I would say account is appropriate here, since an account can be unstaked, yet still have a reward remaining for example - so, in fact, not all of the accounts are Provisioners.

Reply:
You seem to assume a reward can be withdrawn after the unstake, which, tbh, I wasn't considering...
But if that's the case, then after an unstake, that "account" is not a stake anymore.

From what I can see, you consider provisioners as account in the stake contract, whose balance is split into the stake and the reward. If that is the case, then this should be reflected by changing all variables using `stake` as synonym of the StakeContract's account, and only use it to refer to the actual staked amount.

Also, I still think that within the Stake Contract, the keys holding an account should be referred to as Provisioners

Reply:
A reward can indeed be withdrawn after an unstake.

As long as the stake contract also handles rewards, the accounts it handles will not only be Provisioners. The best we can do about this is take the separation of rewards from the stake data (different map), in a different PR.

Reply:
For the sake of completeness, I think account will always be Provisioners, being it they currently have a stake or not.
I'm saying this because rewards are only given to Provisioners.